File: app/src/main/java/com/michaelfotiadis/locationmanagerviewer/LocationApplication.java
package com.michaelfotiadis.locationmanagerviewer;

import android.app.Application;

import com.google.firebase.FirebaseApp;
import com.michaelfotiadis.locationmanagerviewer.data.datastore.Singleton;
import com.michaelfotiadis.locationmanagerviewer.utils.AppLog;

public class LocationApplication extends Application {
    @Override
    public void onCreate() {
        AppLog.i("Singleton Initialised with App Context");
        super.onCreate();
        Singleton.getInstance().setContext(this.getApplicationContext());
        FirebaseApp.initializeApp(this);
    }
}

File: app/src/main/java/com/michaelfotiadis/locationmanagerviewer/LocationApplication.kt
package com.michaelfotiadis.locationmanagerviewer

import android.app.Application
import com.michaelfotiadis.locationmanagerviewer.data.ApplicationInitializer
import com.michaelfotiadis.locationmanagerviewer.injection.appModule
import com.michaelfotiadis.locationmanagerviewer.injection.homeActivityModule
import com.michaelfotiadis.locationmanagerviewer.injection.locationModule
import org.koin.android.ext.android.inject
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin
import org.koin.core.logger.Level

@Suppress("unused")
class LocationApplication : Application() {

    private val applicationInitializer: ApplicationInitializer by inject()

    override fun onCreate() {
        super.onCreate()

        startKoin {
            androidLogger(level = Level.DEBUG)
            androidContext(this@LocationApplication)
            modules(
                appModule,
                locationModule,
                homeActivityModule
            )
        }

        applicationInitializer.run {
            initialiseLogging()
            initialiseFirebase()
        }

    }
}